app/models/rook.rb

```diff
@@ -1,3 +1,10 @@
 class Rook < ChessPiece
 
+  def valid_move?(endpoint_x, endpoint_y)
+    if diagonal_move?(endpoint_x, endpoint_y)
+      return false
+    end
+    is_obstructed?(endpoint_x, endpoint_y)
```

Comment:
let's keep this a separate method so the caller can decide what methods to call.